File: frontend/src/api/linkbio.ts
import apiClient, { unwrapResponse } from './client'
import type { ResData } from '@/types/api'

export interface LinkBioLinkResponse {
  id: number
  title: string
  url: string
  icon: string | null
  sortOrder: number
  clickCount: number
  isActive: boolean
  createdAt: string
}

export interface LinkBioPageResponse {
  id: number
  slug: string
  title: string | null
  bio: string | null
  avatarUrl: string | null
  theme: string
  backgroundColor: string
  textColor: string
  isPublished: boolean
  viewCount: number
  links: LinkBioLinkResponse[]
  createdAt: string
  updatedAt: string
}

export interface CreatePageRequest {
  slug: string
  title?: string
  bio?: string
  avatarUrl?: string
  theme?: string
  backgroundColor?: string
  textColor?: string
}

export interface UpdatePageRequest {
  slug?: string
  title?: string
  bio?: string
  avatarUrl?: string
  theme?: string
  backgroundColor?: string
  textColor?: string
}

export interface LinkItem {
  id?: number
  title: string
  url: string
  icon?: string
  sortOrder: number
  isActive: boolean
}

export interface LinkBioPublicResponse {
  slug: string
  title: string | null
  bio: string | null
  avatarUrl: string | null
  theme: string
  backgroundColor: string
  textColor: string
  links: { id: number; title: string; url: string; icon: string | null; sortOrder: number }[]
}

export interface LinkBioAnalyticsResponse {
  viewCount: number
  links: { id: number; title: string; clickCount: number }[]
}

export const linkBioApi = {
  getPage() {
    return apiClient.get<ResData<LinkBioPageResponse | null>>('/linkbio').then(unwrapResponse)
  },

  createPage(request: CreatePageRequest) {
    return apiClient.post<ResData<LinkBioPageResponse>>('/linkbio', request).then(unwrapResponse)
  },

  updatePage(request: UpdatePageRequest) {
    return apiClient.put<ResData<LinkBioPageResponse>>('/linkbio', request).then(unwrapResponse)
  },

  updateLinks(links: LinkItem[]) {
    return apiClient.put<ResData<LinkBioPageResponse>>('/linkbio/links', { links }).then(unwrapResponse)
  },

  togglePublish(isPublished: boolean) {
    return apiClient.put<ResData<LinkBioPageResponse>>('/linkbio/publish', { isPublished }).then(unwrapResponse)
  },

  getPublicPage(slug: string) {
    return apiClient.get<ResData<LinkBioPublicResponse>>(`/linkbio/${slug}`).then(unwrapResponse)
  },

  getAnalytics() {
    return apiClient.get<ResData<LinkBioAnalyticsResponse>>('/linkbio/analytics').then(unwrapResponse)
  },
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/linkbio/LinkBio.kt
package com.ongo.domain.linkbio

import java.time.LocalDateTime

data class LinkBioPage(
    val id: Long? = null,
    val userId: Long,
    val slug: String,
    val title: String? = null,
    val bio: String? = null,
    val avatarUrl: String? = null,
    val theme: String = "default",
    val backgroundColor: String = "#ffffff",
    val textColor: String = "#000000",
    val isPublished: Boolean = false,
    val viewCount: Long = 0,
    val createdAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime? = null,
)

data class LinkBioLink(
    val id: Long? = null,
    val pageId: Long,
    val title: String,
    val url: String,
    val icon: String? = null,
    val sortOrder: Int = 0,
    val clickCount: Long = 0,
    val isActive: Boolean = true,
    val createdAt: LocalDateTime? = null,
)
